zerver/lib/markdown/__init__.py

```diff
@@ -1060,6 +1119,11 @@ def handle_image_inlining(
         self.add_a(info['parent'], actual_url, url, title=info['title'], insertion_index=info['index'])
         if info['remove'] is not None:
             info['parent'].remove(info['remove'])
+        # Default img generated by ![title](url) markdown can't be
+        # used because we use div tag with images and div inside p
+        # is not correct. So, we remove the img tags and use our
+        # own version of imgs.
```

Comment:
This approach confuses me. There should be no `div`s involved. CommonMark converts

```md
Here is an image: ![title](url)
```

into

```html
<p>Here is an image: <img src="url" title="title" /></p>
```

and that is exactly what I want us to do. If we have legacy code that tries to stick `div`s into this for some legacy reason, that code should be mercilessly deleted. What am I missing?

Reply:
Makes sense, thank you.